### PR TITLE
Add MPIGradient

### LINKED ISSUE #62: Gradient
I suggest to finalize the suite of derivative operators but adding also Gradient as the MPI version of pylops.Gradient.

This should be almost identical to Laplacian with the only difference that we will stack various first derivatives instead of summing them… the only thing I’m not so sure is if our VStack is ready/suitable yet to stacking other pylops_mpi operators? And if this is not the case, how easy it would be to enable this pattern?

### PR DESCRIPTION
**Pull request by rohanbabbar04**
closes #62 
 - Add docstring to `matvec` and `rmatvec` in StackedLinearOperator.
 - Add `MPIGradient`
 - Add a test in `test_derivative.py`
 - Add an example in `plot_derivative.py`
 - Update `index.rst` and added MPIGradient in `basicoperators/__init__.py`

### CODE REVIEW

**Comment by mrava87** [comment]
Good stuff @rohanbabbar04 :) I have just started reviewing this and will try to give you all my comments in the next few days

**Review by mrava87 — COMMENTED** [review]
@rohanbabbar04, this looks great :) I just pushed some minor changes and left a comment for you but i think this is very close to be ready to merge. Good job!

File: pylops_mpi/basicoperators/Gradient.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        self.sampling = sampling
+        self.edge = edge
+        self.kind = kind
+        self.base_comm = base_comm
+        self.dtype = np.dtype(dtype)
+        self.Op = self._calc_stack_op(ndims)
+        super().__init__(shape=self.Op.shape, dtype=dtype, base_comm=base_comm)
+
+    def _matvec(self, x: DistributedArray) -> StackedDistributedArray:
+        return self.Op @ x
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply pylops_mpi/basicoperators/Gradient.py]: Done...

File: pylops_mpi/basicoperators/Gradient.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    def _rmatvec(self, x: StackedDistributedArray) -> DistributedArray:
+        return self.Op.H @ x
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply pylops_mpi/basicoperators/Gradient.py]: Done...

**Comment by rohanbabbar04** [comment]
Sure @mrava87 I have done the changes, if it looks good you can merge this...

**Comment by mrava87** [comment]
Great @rohanbabbar04, I am merging this.

Btw, feel free to go ahead and prepare a release :) otherwise, I can try to do so in coming weeks when I have some time...

**Pull request merged**